eslint/babel-eslint-tests/test/helpers/eslintConfigCompat.cjs

```diff
@@ -0,0 +1,42 @@
+const eslint = require("eslint");
+const babelESLintParserPath = require.resolve("../../../babel-eslint-parser");
+/**
+ * Convert ESLint 9 config to ESLint 7/8 config
+ * @param {eslint.Linter.FlatConfig} ESLint 9 config
+ */
+function eslintConfigCompatTo8(config) {
+  const newConfig = { ...config };
+  const { languageOptions } = config;
+  if (languageOptions) {
+    const { parser, parserOptions, globals: globalsOption } = languageOptions;
+    if (parser) {
+      newConfig.parser = babelESLintParserPath;
```

Comment:
Use the resolved parser path instead of `parser.meta.name` as the rule tester in ESLint 7/8 expects `parser` to be an absolute path. We also pass absolute path to `linter.defineParser` so that the transformed config works for both `linter.verify()` and the `RuleTester`.